src/Language/PureScript/Errors.hs

```diff
@@ -789,6 +793,10 @@ prettyPrintSingleError full level showWiki e = flip evalState defaultUnknownMap
       paras [ line $ "An export for " ++ prettyPrintExport x ++ " requires the following to also be exported: "
             , indent $ paras $ map (line . prettyPrintExport) ys
             ]
+    renderSimpleErrorMessage (TransitiveDctorExportError x ctor) =
+      paras [ line $ "An export for " ++ prettyPrintExport x ++ " requires the following data constructor to also be exported: "
+            , indent $ line $ runProperName ctor
+            ]
```

Comment:
I did this with a new error message as we can't really generate a good error message for the export using a `DeclarationRef` and the `TransitiveDctorExportError`, as it would mean doing quite a bit of work to figure out if the type for the data constructor is already exported, and also what data constructors are exported for that type. Also `TransitiveDctorExportError` doesn't print data constructors currently, for some other complicated reasons (data constructors for the types in here are `error` sometimes).